include/range/v3/view/unique.hpp

```diff
@@ -33,27 +33,44 @@ namespace ranges
         {
             struct unique_fn
             {
-                template<typename Rng>
+            private:
+                friend view_access;
+                template<typename C,
+                        CONCEPT_REQUIRES_(!Range<C>())>
+                static auto bind(unique_fn unique, C pred)
+                RANGES_DECLTYPE_AUTO_RETURN
+                (
+                    std::bind(unique, std::placeholders::_1, protect(std::move(pred)))
+                )
+
+            public:
+                template<typename Rng, typename C = equal_to,
+                        typename I = iterator_t<Rng>>
                 using Concept = meta::and_<
                     ForwardRange<Rng>,
-                    EqualityComparable<range_value_type_t<Rng>>>;
+                    IndirectRelation<C, I>>;
 
-                template<typename Rng, CONCEPT_REQUIRES_(Concept<Rng>())>
-                unique_view<all_t<Rng>> operator()(Rng && rng) const
+                template<typename Rng, typename C = equal_to,
+                        CONCEPT_REQUIRES_(Concept<Rng, C>())>
+                auto operator()(Rng && rng, C pred = equal_to{}) const ->
+                adjacent_filter_view<all_t<Rng>, decltype(not_fn(pred))>
                 {
-                    return {all(static_cast<Rng&&>(rng)), not_equal_to{}};
+                    return {all(static_cast<Rng&&>(rng)), not_fn(pred)};
                 }
             #ifndef RANGES_DOXYGEN_INVOKED
-                template<typename Rng,
-                    CONCEPT_REQUIRES_(!Concept<Rng>())>
+                template<typename Rng, typename C,
+                    CONCEPT_REQUIRES_(!Concept<Rng, C>())>
                 void operator()(Rng &&) const
                 {
                     CONCEPT_ASSERT_MSG(ForwardRange<Rng>(),
                         "The object on which view::unique operates must be a model the "
                         "ForwardRange concept.");
-                    CONCEPT_ASSERT_MSG(EqualityComparable<range_value_type_t<Rng>>(),
+                    using I = iterator_t<Rng>;
+                    CONCEPT_ASSERT_MSG(IndirectRelation<C, I>(),
```

Comment:
Unlike on line 48, compile time isn't an issue here - but it would save a line to use `iterator_t<Rng>` directly here.